tools/dump_tool/dump_tool.c

```diff
@@ -290,13 +294,78 @@ static int ramdump_recv(int dev_fd)
 				}
 			}
 			printf("]\n");
+			fflush(stdout);
 			fclose(bin_fp);
 		}
 	}
 	return 0;
 
 }
 
+static int filedump_recv(int dev_fd)
+{
+	int ret;
+	uint32_t smartfs_addr;
+	uint32_t smartfs_size;
+	char fs_bin[BINFILE_NAME_SIZE] = { '\0' };
+	FILE *fs_fp;
+	char buf;
+	uint32_t count = 0;
+
+	/* Static for now - ARTIK, to be made dynamic */
+	smartfs_addr = 0x04620000;
```

Comment:
This needs to be modified.
Unless you receive the address & size from the  target, this mechanism is wrong.

Reply:
Updated